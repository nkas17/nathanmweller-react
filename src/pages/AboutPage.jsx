import React from 'react';
import Header from '../components/Header';
import Footer from '../components/Footer';

const AboutPage = () => [
	<Header key="about-1" />,
	<main className="main flex-wrapper" key="about-2">
		<div className="flex-item main__flex-item">
			<header className="main__header">
				<h2 className="header header_size_large">
					about<span className="dilbert">&nbsp;</span>
				</h2>
			</header>
			<article className="article flex-wrapper">
				<div className="flex-item article__content">
					<header className="main__header article__header">
						<h3 className="header header_size_medium">experience</h3>
					</header>
					<p className="article__paragraph">
						13+ years software development, 9+ years web development
					</p>
					<header className="main__header article__header">
						<h3 className="header header_size_medium">web development</h3>
					</header>
					<p className="article__paragraph">
						JavaScript, ES2015, Node.js, npm, React.js, Redux.js, Jest, Enzyme,
						Lodash, jQuery, Backbone.js, Underscore.js, HTML, CSS, Sass,
						Bootstrap, responsive design, Google Analytics, Git, PRPC
					</p>
					<header className="main__header article__header">
						<h3 className="header header_size_medium">education</h3>
					</header>
					<p className="article__paragraph">
						BS in Computer Science with honors from Michigan State University
					</p>
					<p className="article__paragraph">
						<a className="link" href="https://www.codeschool.com/users/2817163">
							Code School Badges
						</a>
					</p>
					<header className="main__header article__header">
						<h3 className="header header_size_medium">creations</h3>
					</header>
					<p className="article__paragraph">
						<a className="link" href="https://www.games.nathanmweller.com">
							games
						</a>
						, who doesn't love games
					</p>
					<p className="article__paragraph">
						<a className="link" href="https://www.recipes.nathanmweller.com">
							recipes
						</a>
					</p>
					<p className="article__paragraph">
						<a
							className="link"
							href="https://www.local-weather.nathanmweller.com"
						>
							local weather
						</a>
					</p>
					<p className="article__paragraph">
						<a
							className="link"
							href="https://www.random-quote.nathanmweller.com"
						>
							random quote
						</a>
					</p>
				</div>
				<Footer />
			</article>
		</div>
	</main>,
];

export default AboutPage;
