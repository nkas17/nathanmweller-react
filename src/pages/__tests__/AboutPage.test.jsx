import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AboutPage from '../AboutPage';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <AboutPage />
      </MemoryRouter>,
    )

    .toJSON();
  expect(tree).toMatchSnapshot();
});
