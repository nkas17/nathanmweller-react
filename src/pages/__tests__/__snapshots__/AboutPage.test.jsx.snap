// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="page-header flex-wrapper"
  >
    <div
      className="personal-info page-header__flex-item flex-item"
    >
      <h1
        className="page-header__title"
      >
        nathan m weller
      </h1>
      <p
        className="page-header__description"
      >
        husband, father, software architect, web developer, cook, photographer, fútbol player, gamer
      </p>
      <ul
        className="contact-info page-header__contact-info"
      >
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="mailto:nathan@nathanmweller.com?subject=Reaching%20Out"
          >
            <i
              className="fa fa-envelope-o"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="https://www.linkedin.com/pub/nathan-weller/52/b05/552"
          >
            <i
              className="fa fa-linkedin"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="https://twitter.com/nkas17_m"
          >
            <i
              className="fa fa-twitter"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="https://plus.google.com/u/0/100022763393186980494/posts/p/pub"
          >
            <i
              className="fa fa-google-plus"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="https://www.facebook.com/nathan.weller.9"
          >
            <i
              className="fa fa-facebook"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="tel:6168430170"
          >
            <i
              className="fa fa-phone"
            />
          </a>
        </li>
        <li
          className="contact-info__item"
        >
          <a
            className="link"
            href="sms:6168430170"
          >
            <i
              className="fa fa-comment-o"
            />
          </a>
        </li>
      </ul>
    </div>
    <nav
      className="nav flex-item"
    >
      <ul>
        <li
          className="nav__item"
        >
          <a
            className="link nav__link"
            href="/"
            onClick={[Function]}
          >
            home
          </a>
        </li>
        <li
          className="nav__item"
        >
          <a
            className="link nav__link"
            href="/fed"
            onClick={[Function]}
          >
            fed
          </a>
        </li>
        <li
          className="nav__item"
        >
          <a
            className="link nav__link"
            href="https://www.blog.nathanmweller.com"
          >
            blog
          </a>
        </li>
        <li
          className="nav__item"
        >
          <a
            className="link nav__link"
            href="/about"
            onClick={[Function]}
          >
            about
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="logo flex-item"
    >
      <img
        alt="nathan m weller"
        className="logo__image"
        height="200"
        src="test-file-stub"
        width="200"
      />
    </div>
  </header>,
  <main
    className="main flex-wrapper"
  >
    <div
      className="flex-item main__flex-item"
    >
      <header
        className="main__header"
      >
        <h2
          className="header header_size_large"
        >
          about
          <span
            className="dilbert"
          >
             
          </span>
        </h2>
      </header>
      <article
        className="article flex-wrapper"
      >
        <div
          className="flex-item article__content"
        >
          <header
            className="main__header article__header"
          >
            <h3
              className="header header_size_medium"
            >
              experience
            </h3>
          </header>
          <p
            className="article__paragraph"
          >
            13+ years software development, 9+ years web development
          </p>
          <header
            className="main__header article__header"
          >
            <h3
              className="header header_size_medium"
            >
              web development
            </h3>
          </header>
          <p
            className="article__paragraph"
          >
            JavaScript, ES2015, Node.js, npm, React.js, Redux.js, Jest, Enzyme, Lodash, jQuery, Backbone.js, Underscore.js, HTML, CSS, Sass, Bootstrap, responsive design, Google Analytics, Git, PRPC
          </p>
          <header
            className="main__header article__header"
          >
            <h3
              className="header header_size_medium"
            >
              education
            </h3>
          </header>
          <p
            className="article__paragraph"
          >
            BS in Computer Science with honors from Michigan State University
          </p>
          <p
            className="article__paragraph"
          >
            <a
              className="link"
              href="https://www.codeschool.com/users/2817163"
            >
              Code School Badges
            </a>
          </p>
          <header
            className="main__header article__header"
          >
            <h3
              className="header header_size_medium"
            >
              creations
            </h3>
          </header>
          <p
            className="article__paragraph"
          >
            <a
              className="link"
              href="https://www.games.nathanmweller.com"
            >
              games
            </a>
            , who 
            doesn't
             love games
          </p>
          <p
            className="article__paragraph"
          >
            <a
              className="link"
              href="https://www.recipes.nathanmweller.com"
            >
              recipes
            </a>
          </p>
          <p
            className="article__paragraph"
          >
            <a
              className="link"
              href="https://www.local-weather.nathanmweller.com"
            >
              local weather
            </a>
          </p>
          <p
            className="article__paragraph"
          >
            <a
              className="link"
              href="https://www.random-quote.nathanmweller.com"
            >
              random quote
            </a>
          </p>
        </div>
        <footer
          className="footer flex-item"
        >
          <div
            className="logo footer__logo"
          >
            <img
              alt="nathan m weller"
              className="logo__image"
              height="200"
              src="test-file-stub"
              width="200"
            />
            <p
              className="logo__description"
            >
              husband, father, software architect, web developer, cook, photographer, fútbol player, gamer
            </p>
          </div>
        </footer>
      </article>
    </div>
  </main>,
]
`;
